_posts/2023-05-17-otp-26-highlights.md

```diff
@@ -0,0 +1,892 @@
+4> lists:enumerate(0, -1, [a,b,c]).
+[{0,a},{-1,b},{-2,c}]
+```
+
+## New options for the `zip` family of functions
+
+The `zip` family of functions in the `lists` modules combines two or three lists
+into a single list of tuples. For example:
+
+```erlang
+1> lists:zip([a,b,c], [1,2,3]).
+[{a,1},{b,2},{c,3}]
+
+```
+
+The existing `zip` functions fail if the lists don't have the same length:
+
+```erlang
+2> lists:zip([a,b,c,d], [1,2,3]).
+** exception error: no function clause matching . . .
+```
+
+In OTP 26, the `zip` functions now take an extra `How` parameter that
+determines what should happen when the lists are of unequal length.
+
+For some use cases for `zip`, ignoring the superfluous elements in the
+longer list or lists can make sense. That can be done using the `trim`
+option:
+
+```erlang
+3> lists:zip([a,b,c,d], [1,2,3], trim).
+[{a,1},{b,2},{c,3}]
+```
+
+For other use cases it could make more sense to extend the shorter
+list or lists to the length of the longest list. That can be done
+using the `{pad, Defaults}` option, where `Defaults` should be a tuple
+having the same number of elements as the number of lists. For
+`lists:zip/3`, that means that the `Defaults` tuple should have two
+elements:
+
+```erlang
+4> lists:zip([a,b,c,d], [1,2,3], {pad, {zzz, 999}}).
+[{a,1},{b,2},{c,3},{d,999}]
+5> lists:zip([a,b,c], [1,2,3,4,5], {pad, {zzz, 999}}).
+[{a,1},{b,2},{c,3},{zzz,4},{zzz,5}]
+```
+
+For `lists:zip3/3` the `Defaults` tuple should have three elements:
+
+```erlang
+6> lists:zip3([], [a], [1,2,3], {pad, {0.0, zzz, 999}}).
+[{0.0,a,1},{0.0,zzz,2},{0.0,zzz,3}]
+```
+
+# No need to enable feature `maybe` in the runtime system
+
+In OTP 25, the [feature
+concept](https://www.erlang.org/doc/reference_manual/features.html)
+and the [maybe
+feature](https://www.erlang.org/doc/reference_manual/expressions.html#maybe)
+were introduced. In order to use `maybe` in OTP 25, is is necessary to
+enable it in both the compiler and the runtime system. For example:
+
+```
+$ cat t.erl
+-module(t).
+-feature(maybe_expr, enable).
+-export([listen_port/2]).
+listen_port(Port, Options) ->
+    maybe
+        {ok, ListenSocket} ?= inet_tcp:listen(Port, Options),
+        {ok, Address} ?= inet:sockname(ListenSocket),
+        {ok, {ListenSocket, Address}}
+    end.
+$ erlc t.erl
+$ erl
+Erlang/OTP 25 . . .
+
+Eshell V13.1.1  (abort with ^G)
+1> t:listen_port(50000, []).
+=ERROR REPORT==== 6-Apr-2023::12:01:20.373223 ===
+Loading of . . ./t.beam failed: {features_not_allowed,
+                                 [maybe_expr]}
+
+** exception error: undefined function t:listen_port/2
+2> q().
+$ erl -enable-feature maybe_expr
+Erlang/OTP 25 . . .
+
+Eshell V13.1.1  (abort with ^G)
+1> t:listen_port(50000, []).
+{ok,{#Port<0.5>,{{0,0,0,0},50000}}}
+```
+
+In OTP 26, it is no longer necessary to enable a feature in the
+runtime system in order to load modules that are using it. It is
+sufficient to enable the feature in the compiler when compiling it.
```

Comment:
"It is sufficient to enable the feature in the compiler when compiling it."

If I had not debugged a crash from private_append which was triggered by an example using `maybe` i would look for a flag to `erlc` in the example below.

"It is sufficient to have `-feature(maybe_expr, enable).` in the module." perhaps?